Parse/Internal/User/AuthenticationProviders/Controller/PFUserAuthenticationController.m

```diff
@@ -56,9 +57,11 @@ - (void)registerAuthenticationDelegate:(id<PFUserAuthenticationDelegate>)delegat
     });
 
     // TODO: (nlutsenko) Decouple this further.
-    if (![authType isEqualToString:PFAnonymousUserAuthenticationType]) {
-        [[PFUser currentUser] synchronizeAuthDataWithAuthType:authType];
-    }
+    [[[PFUser currentUserController] getCurrentUserAsyncWithOptions:0] continueWithSuccessBlock:^id(BFTask *task) {
```

Comment:
How hard is it to have this injected? I don't really like having this be harder than necessary to test.